File: selection-sort/selection.py
def selection_sort(array):
    for i in range(0, len(array)) :
        minIndex = i
        for j in range(i, len(array)):
            if array[j] < array[minIndex]:
                minIndex = j
        if minIndex != i:
            temp = array[i]
            array[i] = array[minIndex]
            array[minIndex] = temp
    return array

print(selection_sort([5,2,1,50,23,12,66,8]))

File: selection-sort/selection.js
// Time complexity O(n^2)
function selectionSort(array) { 
    
    for(let i= 0; i < array.length; i++){
      let minIndex = i;
  
      for (let j = i ; j < array.length ; j++){
        if (array[j]<array[minIndex]){
          minIndex = j;
        }
      }
      if( minIndex !== i){
        let temp = array[i];
        array[i] = array[minIndex];
        array[minIndex]=temp;
      }
    }
    
    return array;
  }
  console.log(selectionSort([5,2,1,50,23,12]))
